### PR TITLE
Support non-DSN based ODBC connection.

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

- Support ODBC connection with a host, port, database, user, and password
- Support ODBC connection with a connection string.

# Checklist
Make sure you have performed the following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/system.R [hunk 1]
```diff
@@ -1176,7 +1176,67 @@ getDBConnection <- function(type, host = NULL, port = "", databaseName = "", use
...
         conn <- eval(parse(text=connstr))
       } else if (host != "") {
-        # TODO: Implement direct connect
+        connectionString <- stringr::str_c(
+          "Driver=", driver, ";Server=", host, ";Port=", port, ";Database=", databaseName,
+          ";UID=", username, ";PWD=", password
+        );
+        if (additionalParams != "") {
+          # The "additionalParams" is passed as 'a=1,b=2,c=3'. Replace"," with ";" so that connection string because a=1;b=2;c=3
+          connectionString <- stringr::str_c(connectionString, ";", stringr::str_replace(additionalParams,",", ";"));
```
Comment (hideaki) [inline R/system.R:1185]: Please clean up the spacing.

File: R/system.R [hunk 2]
```diff
@@ -1176,7 +1176,67 @@ getDBConnection <- function(type, host = NULL, port = "", databaseName = "", use
         }
         conn <- eval(parse(text=connstr))
       } else if (host != "") {
-        # TODO: Implement direct connect
+        connectionString <- stringr::str_c(
+          "Driver=", driver, ";Server=", host, ";Port=", port, ";Database=", databaseName,
```
Comment (hideaki) [inline R/system.R:1180]: Are these keywords like Server, UID, PWD really generic for any ODBC driver?

File: R/system.R [hunk 3]
```diff
@@ -1176,7 +1176,67 @@ getDBConnection <- function(type, host = NULL, port = "", databaseName = "", use
...
+          ";UID=", username, ";PWD=", password
+        );
+        if (additionalParams != "") {
+          # The "additionalParams" is passed as 'a=1,b=2,c=3'. Replace"," with ";" so that connection string because a=1;b=2;c=3
```
Comment (hideaki) [inline R/system.R:1184]: Please correct the grammatical error in the comment.

File: R/system.R [hunk 4]
```diff
@@ -1176,7 +1176,67 @@ getDBConnection <- function(type, host = NULL, port = "", databaseName = "", use
         }
         conn <- eval(parse(text=connstr))
       } else if (host != "") {
```
Comment (hideaki) [inline R/system.R:1178]: Check subtype instead, which should take values "dsn", "conn_str_kv", or "conn_str_text".

Comment (hidekoji) [reply R/system.R]: Updated as suggested.

File: R/system.R (hunk 3 above)
Comment (hidekoji) [reply R/system.R]: Fixed grammar error.

**Pull request merged**